test/e2e/e2e_suite_test.go

```diff
@@ -493,7 +493,7 @@ func getLatestSkuForMinor(version string, skus semver.Versions) string {
 				break
 			}
 		}
-	} else {
+	} else if strings.HasPrefix(version, "v") {
```

Comment:
What are the possible values for version here?  

v1.21.0
latest-1.21

Reply:
it could be either `vx.y.z` (eg. v1.21.0), `latest` or `latest-x.y` (eg. latest-1.21)

https://capz.sigs.k8s.io/developers/development.html#conformance-testing

If it's a latest- version (or anything else that could be added in the future), we don't check for a valid semver and an existing VM image and instead keep the previous behavior.

Reply:
actually, wait

should I just change it back to what I had here originally: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1767#discussion_r749648252

/hold

Reply:
/hold cancel

@jsturtevant check it out, I think this is better

Reply:
the comments help alot thanks